### PR TITLE
New Nonlinear Audio Bus Fader

### PR DESCRIPTION
**Pull request by Eoin-ONeill-Yokai**
This patch changes the audio mixer faders to use a non-linear algorithm
for volume control. 

godot-audio-patch-s0

## Why use a non-linear fader?
The intention is to make Godot's audio faders be more like those found in professional audio equipment and programs. It is an exponential equation which intends to counter-act the logarithmic nature of human hearing. **The effect of this change is a more usable audio mixer with more emphasis on the values that make the most difference to the mix.** I found [this](https://www.dr-lex.be/info-stuff/volumecontrols.html) website to be a decent explanation for why this is necessary. You can also observe this type of non-linear meter in programs such as Ableton Live or Bitwig Studio.

### The Math
The scaling algorithm is composed of three hand-made equations. Two of which are simple linear equations which intend to ease off the exponential behaviour toward the ends of the fader. The last equation is an exponential equation that counteracts the logarithmic nature of human hearing.

This patch also changes the audio level notch widget to be less static and thus supports changing the scaling factor of the audio faders. This should make it easier to change the fader scale in the future, including to revisions done to this patch. This should make it easier for us to work together on a better algorithm is necessary since the groundwork is now in place to make changes as we see fit without having to worry about creating a new texture to reflect the fader changes. 

### One last thing...
godot-audio-patch-s1
This patch also adds a value preview when you change the fader. It will show you the value of fader as you change it which provides a decent visual feedback to changing the value. There's a timer which controls how long the preview stays after a value change.

### CODE REVIEW

**Comment by samgreen** [comment]
I fully support this change

**Comment by reduz** [comment]
this looks great, feel free to merge

File: editor/editor_audio_buses.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*************************************************************************/
+﻿/*************************************************************************/
```
Comment (akien-mga) [inline editor/editor_audio_buses.cpp]: It looks like your IDE added an unnecessary BOM (Byte Order Mark) here. Could you amend the commit to remove it?

Comment (TheYokai) [reply editor/editor_audio_buses.cpp]: The BOM should be removed. Sorry about that, I'm not sure how that got there.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**